doc/source/dev/toolchain.rst

```diff
@@ -413,7 +413,7 @@ Building the Documentation
  Tool                 Version
 ====================  =================================================
 Sphinx                Whatever recent versions work. >= 2.0.
-PyData Sphinx theme   Whatever recent versions work. >= 0.8.1.
+PyData Sphinx theme   Whatever recent versions work. >= 0.15.1.
 Sphinx-Design         Whatever recent versions work. >= 0.2.0.
 numpydoc              Whatever recent versions work. >= 0.8.0.
 matplotlib            Generally suggest >= 2.0.
```

Comment:
Can we update the dependencies here? For example, the new theme [requires](https://github.com/pydata/pydata-sphinx-theme/blob/v0.15.1/pyproject.toml#L23) at least sphinx 5. I'm surprised it has no lower bound on sphinx-design, but with the docs build [using](https://github.com/scipy/scipy/blob/7ff74ee275de307d6a17bebff4ea10cf58d35dff/.circleci/config.yml#L119) python 3.11, it [seems](https://github.com/executablebooks/sphinx-design/releases) we need at least 0.4.

And so on (matplotlib 2 is freaking ancient, should probably require 3.x for some recent x)